src/lib/lineage-manipulation.js

```diff
@@ -0,0 +1,72 @@
+
+/*
+Given a doc, replace the lineage information therein with "replaceWith"
+
+startingFromIdInLineage (optional) - Will result in a partial replacement of the lineage. Only the part of the lineage "after" the parent
+with _id=startingFromIdInLineage will be replaced by "replaceWith"
+*/
+const replaceLineage = (doc, replaceWith, startingFromIdInLineage) => {
+  const lineageAttributeName = doc.type === 'data_record' ? 'contact' : 'parent';
```

Comment:
I've updated to handle the `contact` attribute.

Just for my education: is this in the current version or only older versions? 

* In master I can't get a lineage to appear for the `contact` attribute
* Looking at Muso data (3.3) I do see a lineage because the "contact" attribute is a full copy of the primary contact
* Looking at LG-AN (2.16), they have minified lineages

Reply:
I think it depends on the xform (?) or maybe it's a bug. 
I've also noticed that when you create the primary contact along with the place, it just saves the id, but when you assign another contact as primary on edit, it saves full lineage. 
Maybe we should open an issue just to clarify this.

Reply:
https://github.com/medic/medic/issues/5715